storage/remote/write_handler.go

```diff
@@ -610,18 +622,18 @@ func (rw *rwExporter) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) er
 		LookbackDelta:                     rw.lookbackDelta,
 		EnableTypeAndUnitLabels:           rw.enableTypeAndUnitLabels,
 	})
-	if err != nil {
-		rw.logger.Warn("Error translating OTLP metrics to Prometheus write request", "err", err)
-	}
+
+	defer func() {
+		if err != nil {
+			_ = app.Rollback()
+			return
+		}
+		err = app.Commit()
```

Comment:
Isn't this commiting twice? (I see combinedAppender.Commit invoked in converter code)